pilot/pkg/serviceregistry/kube/controller/ambient/workloads.go

```diff
@@ -420,20 +420,22 @@ func (a *index) serviceEntryWorkloadBuilder(
 
 			policies := a.buildWorkloadPolicies(ctx, authorizationPolicies, peerAuths, meshCfg, se.Labels, se.Namespace)
 
-			var waypointAddress *workloadapi.GatewayAddress
+			var waypoint *Waypoint
 			// Endpoint does not have a real ObjectMeta, so make one
 			if !implicitEndpoints {
-				if waypoint, err := fetchWaypointForWorkload(ctx, waypoints, namespaces, metav1.ObjectMeta{
+				if wp, err := fetchWaypointForWorkload(ctx, waypoints, namespaces, metav1.ObjectMeta{
 					Name:      se.Name,
 					Namespace: se.Namespace,
 					Labels:    wle.Labels,
-				}); err != nil {
-					waypointAddress = waypoint.Address
-					// enforce traversing waypoints
-					policies = append(policies, implicitWaypointPolicies(ctx, waypoints, waypoint, services)...)
+				}); err == nil {
```

Comment:
should we log error?

Reply:
I'd actually argue we may want to call this variable something like "status" rather than "err" just to break the automatic `err != nil` type of thinking.